### PR TITLE
Enums for nodejs sdk

### PR DESCRIPTION
**Pull request by amahadaya**
Generation of the enum and modified model files from the edited mustache template files

### CODE REVIEW

File: __tests__/DefaultApi.spec.ts [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+import * as freeclimb from '../index'
+import { AccountRequest, AccountResult, ApplicationList, ApplicationRequest, ApplicationResult, AvailableNumberList, BuyIncomingNumberRequest, CallList, CallResult, ConferenceList, ConferenceParticipantList, ConferenceParticipantResult, ConferenceResult, CreateConferenceRequest, FilterLogsRequest, IncomingNumberList, IncomingNumberRequest, IncomingNumberResult, LogList, MessageResult, MessagesList, QueueList, QueueMember, QueueMemberList, QueueRequest, QueueResult, RecordingList, RecordingResult, ServerConfiguration, UpdateCallRequest, UpdateConferenceRequest } from "../index";
+import { AccountStatus } from "../models/AccountStatus";
+import { AccountType } from "../models/AccountType";
+import { CallStatus } from "../models/CallStatus";
+import { MessageDirection } from "../models/MessageDirection";
+import { PlayBeep } from "../models/PlayBeep";
+import { UpdateCallRequestStatus } from "../models/UpdateCallRequestStatus";
+import { UpdateConferenceRequestStatus } from "../models/UpdateConferenceRequestStatus";
+import { HttpFile } from "../index";
```
Comment (ngarg-vail) [inline __tests__/DefaultApi.spec.ts:11]: Can we consolidate these import lines? 
```
import * as freeclimb, { ... } from '../'
```

File: __tests__/DefaultApi.spec.ts [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+        const callId = "callId_example"
+        return apiInstance.dequeueAMember(queueId, callId).then((data) => expect(data).toBeInstanceOf(QueueMember))
+    });
+
+    test('#dequeueHeadMember', async () => {
+        const queueId = "queueId_example"
+        return apiInstance.dequeueHeadMember(queueId).then((data) => expect(data).toBeInstanceOf(QueueMember))
+    });
+
+    //Error on this test due to not being able to find an assertion for httpFile being returned
```
Comment (ngarg-vail) [inline __tests__/DefaultApi.spec.ts:110]: remove

File: __tests__/DefaultApi.spec.ts [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+    test('#sendAnSmsMessage', async () => {
+        const request = {
+            _from: "_from_example",
+            to: "to_example",
+            text: "Example Text"
+        }
+        return apiInstance.sendAnSmsMessage(request).then((data) => expect(data).toBeInstanceOf(MessageResult))
+    });
+
+    //Error on this test due to not being able to find an assertion for httpFile being returned
```
Comment (ngarg-vail) [inline __tests__/DefaultApi.spec.ts:344]: remove

File: __tests__/DefaultApi.spec.ts [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+    });
+
+    test('#updateAnIncomingNumber', async () => {
+        const phoneNumberId = "phoneNumberId_example";
+        const incomingNumberRequest: IncomingNumberRequest  =  {
+            applicationId: "applicationId_example",
+            alias: "alias_example",
+        }
+        return apiInstance.updateAnIncomingNumber(phoneNumberId, incomingNumberRequest).then((data) => expect(data).toBeInstanceOf(IncomingNumberResult))
+    });
```
Comment (ngarg-vail) [inline __tests__/DefaultApi.spec.ts:409]: tests from line 83 no longer contain `expect.assertions(..)` is that intentional?

File: __tests__/percl.spec.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import { describe, expect, test } from "@jest/globals";
-import { Say, GetDigits, PerclScript, Pause, Redirect } from "../";
+import { Say, GetDigits, PerclScript, Pause, Redirect } from "../dist";
```
Comment (ngarg-vail) [inline __tests__/percl.spec.ts:2]: shouldn't need to change this import

File: models/CreateConferenceRequest.ts [hunk 6]
```diff
@@ -70,12 +62,11 @@ export class CreateConferenceRequest {
...
             "name": "playBeep",
             "baseName": "playBeep",
-            "type": "CreateConferenceRequestPlayBeepEnum",
+            "type": "PlayBeep",
             "format": "",
             
-            "defaultValue": CreateConferenceRequestPlayBeepEnum.ALWAYS
             
-
+            "defaultValue": undefined
```
Comment (ngarg-vail) [inline models/CreateConferenceRequest.ts:69]: shouldn't defaultValue be provided?

**Review by ngarg-vail — CHANGES_REQUESTED** [review]

File: __tests__/DefaultApi.spec.ts (hunk 4 above)
Comment (amahadaya) [reply __tests__/DefaultApi.spec.ts:409]: this depends on whether there is a then statement, some of them do not have a then statement so we cannot expect an assertion, I'll re-add in this line for the ones expecting an assertion

Comment (ngarg-vail) [reply __tests__/DefaultApi.spec.ts:409]: shouldn't all methods have a .then ? 
can we just use async/await instead of async + .then?

**Review by ngarg-vail — APPROVED** [review]

File: __tests__/DefaultApi.spec.ts (hunk 4 above)
Comment (amahadaya) [reply __tests__/DefaultApi.spec.ts:399]: not all methods, because some methods return voids, do we want to assert against voids?

**Review by ngarg-vail — APPROVED** [review]

**Review by ngarg-vail — APPROVED** [review]

**Pull request merged**